activemodel/lib/active_model/errors.rb

```diff
@@ -216,35 +218,16 @@ def keys
       messages.keys
     end
 
-    # Returns an array of error messages, with the attribute name included.
-    #
-    #   person.errors.add(:name, :blank, message: "can't be blank")
-    #   person.errors.add(:name, :not_specified, message: "must be specified")
-    #   person.errors.to_a # => ["name can't be blank", "name must be specified"]
-    def to_a
-      full_messages
-    end
-
-    # Returns the number of error messages.
-    #
-    #   person.errors.add(:name, :blank, message: "can't be blank")
-    #   person.errors.count # => 1
-    #   person.errors.add(:name, :not_specified, message: "must be specified")
-    #   person.errors.count # => 2
-    def count
-      to_a.size
-    end
-
     # Returns +true+ if no errors are found, +false+ otherwise.
     # If the error message is a string it can be empty.
     #
     #   person.errors.full_messages # => ["name cannot be nil"]
     #   person.errors.empty?        # => false
     def empty?
-      all? { |k, v| v && v.empty? && !v.is_a?(String) }
+      size.zero?
```

Comment:
Our test coverage is not 100%. I'd find the commit that introduced it
before assuming it's safe to remove

On Fri, Feb 20, 2015, 2:56 PM Wojciech Wnętrzak <notifications@github.com>
wrote:

> In activemodel/lib/active_model/errors.rb
> <https://github.com/rails/rails/pull/19021#discussion_r25106561>:
>
> >      # Returns +true+ if no errors are found, +false+ otherwise.
> >      # If the error message is a string it can be empty.
> >      #
> >      #   person.errors.full_messages # => ["name cannot be nil"]
> >      #   person.errors.empty?        # => false
> >      def empty?
> > -      all? { |k, v| v && v.empty? && !v.is_a?(String) }
> > +      size.zero?
>
> Actually there is also a flatten call so it still returns the same value
> as before.
>
> irb(main):043:0> {foo: []}.all? { |k, v| v && v.empty? && !v.is_a?(String) }
> => true
> irb(main):044:0> {foo: []}.values.flatten.size.zero?
> => true
>
> There is a difference when error is a {}
>
> irb(main):053:0> {foo: {}}.all? { |k, v| v && v.empty? && !v.is_a?(String) }
> => true
> irb(main):054:0>  {foo: {}}.values.flatten.size.zero?
> => false
>
> Previously size and count returned different results for [] case - size
> #=> 0, count #=> 1.
> There is no test for adding [], {} as error value and no documentation of
> such usage and I don't know why someone would want to add such error.
> Should we support it?
> If yes, what should be the proper output, because now it's totally
> inconsistent.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/19021/files#r25106561>.
>